### PR TITLE
chore: remove CVE cuda image

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Related issue

`#` followed by the number of the issue

## Summary

Fixes FL-1645

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

File: substrafl/remote/register/register.py [hunk 1]
```diff
@@ -73,6 +73,10 @@
 # Install requirements
 RUN python{python_version} -m pip install --no-cache-dir -r requirements.txt
 
+USER root
+RUN apt-get purge -y --auto-remove build-essential *-dev
+USER user
+
```
Comment (ThibaultFy) [inline substrafl/remote/register/register.py:79]: Are we forced to do it in last step ? Do we need these libs to setup the image ?

Comment (guilhem-barthes) [reply substrafl/remote/register/register.py:79]: I think we may need these when we compile python module

Comment (ThibaultFy) [reply substrafl/remote/register/register.py:79]: Okay. And using the root user might raise some alert no ?

Comment (guilhem-barthes) [reply substrafl/remote/register/register.py:79]: What do you mean by alert? What kind? To who?

Comment (ThibaultFy) [reply substrafl/remote/register/register.py:79]: On the builder while building the image ? I'm not sure I just want to check that using a root user while building will not trigger anything 😇

Comment (guilhem-barthes) [reply substrafl/remote/register/register.py:79]: This part (before switching to the new user) is already run as root
```
_CPU_BASE_IMAGE = """
FROM python:{python_version}-slim

# update image
RUN apt-get update -y
"""

_GPU_BASE_IMAGE = """
FROM nvidia/cuda:11.8.0-runtime-ubuntu22.04

# update image & install Python
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update -y\
    && apt-get install -y software-properties-common\
    && add-apt-repository -y ppa:deadsnakes/ppa\
    && apt-get -y upgrade\
    && apt-get install -y python{python_version} python{python_version}-venv python3-pip\
    && apt-get clean\
    && rm -rf /var/lib/apt/lists/*

"""
```
what kind of test would you like to run to check if no warning is raised?

Comment (ThibaultFy) [reply substrafl/remote/register/register.py:79]: Okay you're right :) no that's ok then thanks ! we can just launch an e2e test, like mnist, on it :)

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by ThibaultFy — APPROVED** [review]
Thanks :)

**Pull request merged**